crates/miden-lib/src/transaction/inputs.rs

```diff
@@ -55,6 +55,20 @@ impl TransactionAdviceInputs {
             inputs.add_map_entry(account_id_key, seed.to_vec());
         }
 
+        // if the account is new, insert the storage map entries into the advice provider.
+        if partial_native_acc.is_new() {
+            for storage_map in partial_native_acc.storage().maps() {
+                let map_entries = storage_map
+                    .entries()
+                    .flat_map(|(key, value)| {
+                        value.as_elements().iter().chain(key.as_elements().iter()).copied()
+                    })
```

Comment:
It's interesting that we are building the list from entries represented as `[VALUE, KEY]`. I would have thought `[KEY, VALUE]` would have been more natural.